client.go

```diff
@@ -33,6 +36,10 @@ type Client struct {
 
```

Comment:
We could add `HTTPClient: *http.Client` inside `Client`, making it optional to be changed during client's life.